### DISCUSSION TITLE
What strategy would you recommend for consuming remote parts/modules?

### DISCUSSION THREAD

**Discussion by shanesveller** [hercules-ci/flake-parts#54] [Q&A]
I find myself repeating certain flake-parts modules across similar projects (i.e. I have a body of Rust-specific modules) and would love to get some remote re-use of such shared content. Do you have any ideas/plans to make this possible, or do you have any recommendations for syntax that would work well with the current design when the desired content is part of a separate flake input?

**Comment by srid** [comment]
You can have `imports` reference flake inputs, no?

Example:
- https://github.com/srid/haskell-template (and other Haskell projects) reuse https://github.com/srid/haskell-flake

If you look in the `flake.nix` of haskell-template: https://github.com/srid/haskell-template/blob/dc693ae217e5b8718f2a12c60e02a92f383fc308/flake.nix#L15-L19

```nix
      imports = [
        inputs.haskell-flake.flakeModule
        inputs.treefmt-flake.flakeModule
        inputs.check-flake.flakeModule
      ];
```

All 3 of these modules come from an external flake.
